dao/src/main/scala/za/co/absa/enceladus/dao/EnceladusRestDAO.scala

```diff
@@ -45,7 +60,49 @@ object EnceladusRestDAO extends EnceladusDAO {
     .configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false)
     .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
 
+  private[dao] def getUserFromKeytab(keytabLocation: String): String = {
+    val keytab = KeyTab.getInstance(keytabLocation)
+    keytab.getOneName.getName
+  }
+    
+  def spnegoLogin(keytabLocation: String): Boolean = {
+    import scala.collection.JavaConversions._
+    
+    val principal = getUserFromKeytab(keytabLocation)
+    _userName = principal
+    
+    val template = new KerberosRestTemplate(keytabLocation, principal)
+    val url = s"$restBase/user/info"
+
+    log.info(s"Calling REST with spnego auth $principal $keytabLocation")
+    val response = template.getForEntity(new URI(url), classOf[String])
+
+    val status = response.getStatusCode
+    val ok = status == SpringHttpStatus.OK
+    val unAuthorized = status == SpringHttpStatus.UNAUTHORIZED
+    log.info(response.toString)
+    log.debug(s"OK: $ok")
+    log.debug(s"Unauthorized: $unAuthorized")
+    if (ok) {
```

Comment:
Since this is a duplicate of `postLogin`, I'd extract it.

Reply:
Yeah will do.